packages/@aws-cdk/aws-scheduler-targets-alpha/lib/codepipeline-start-pipeline-execution.ts

```diff
@@ -0,0 +1,41 @@
+import { ISchedule, IScheduleTarget } from '@aws-cdk/aws-scheduler-alpha';
+import { Arn, ArnFormat, Names } from 'aws-cdk-lib';
+import { IPipeline } from 'aws-cdk-lib/aws-codepipeline';
+import { IRole, PolicyStatement } from 'aws-cdk-lib/aws-iam';
+import { ScheduleTargetBase, ScheduleTargetBaseProps } from './target';
+import { sameEnvDimension } from './util';
+
+/**
+ * Use an AWS CodePipeline pipeline as a target for AWS EventBridge Scheduler.
+ */
+export class CodePipelineStartPipelineExecution extends ScheduleTargetBase implements IScheduleTarget {
+  constructor(
+    private readonly pipeline: IPipeline,
+    private readonly props: ScheduleTargetBaseProps = {},
+  ) {
+    super(props, pipeline.pipelineArn);
+  }
+
+  protected addTargetActionToRole(schedule: ISchedule, role: IRole): void {
+    const pipelineArn = Arn.split(this.pipeline.pipelineArn, ArnFormat.NO_RESOURCE_NAME);
+    const region = pipelineArn.region ?? '';
+    const account = pipelineArn.account ?? '';
```

Comment:
Isn't better to parse ARN attributes in [`fromPipelineArn`](https://github.com/aws/aws-cdk/blob/7c497ee998976b69ad1e2cd3e68f30cb593f8ded/packages/aws-cdk-lib/aws-codepipeline/lib/pipeline.ts#L322) directly?